### PR TITLE
Remove unused mdps and scripts

### PR DESCRIPTION
**Pull request by chrishajas**
These are mdps, xmls and some random test files that aren't used and
take up quite a bit of space. I also removed some scripts that we haven't used and haven't touched. Let me know if anything seems worth keeping.

Also, I found a couple of mdps that seem like we intended to test but forgot to add to the file to actually enable the mdp; I've added those and updated them so they pass.

This reduces the directory size from 107MB to 84MB (not including the .git).

### CODE REVIEW

File: server/src/unittest/gpopt/minidump/CICGTest.cpp [hunk 1]
```diff
@@ -62,7 +62,6 @@ const CHAR *rgszFileNames[] =
 
 #ifndef GPOS_DEBUG
 		// TODO:  - Jul 14 2015; disabling it for debug build to reduce testing time
-		// "../data/dxl/minidump/HAWQ-TPCH09-NoTopBroadcast.mdp",
```
Comment (zellerh) [inline server/src/unittest/gpopt/minidump/CICGTest.cpp]: Remove the TODO comment as well?

File: data/dxl/minidump/LeftOuter2InnerUnionAllAntiSemiJoin-Tpcds.mdp [hunk 2]
```diff
@@ -11836,85 +11905,16 @@ select * from v2 left outer join v1 on v2.i_item_sk = v1.ss_item_sk limit 5;
...
-                            <dxl:Cost StartupCost="0" TotalCost="34.916016" Rows="17877.000000" Width="4"/>
-                          </dxl:Properties>
-                          <dxl:ProjList>
-                            <dxl:ProjElem ColId="247" Alias="i_item_sk">
-                              <dxl:Ident ColId="247" ColName="i_item_sk" TypeMdid="0.23.1.0"/>
-                            </dxl:ProjElem>
-                          </dxl:ProjList>
-                        </dxl:CTEConsumer>
-                      </dxl:NestedLoopJoin>
+                      </dxl:CTEConsumer>
```
Comment (ashuka24) [inline data/dxl/minidump/LeftOuter2InnerUnionAllAntiSemiJoin-Tpcds.mdp]: is this a change we expect?

Comment (chrishajas) [reply data/dxl/minidump/LeftOuter2InnerUnionAllAntiSemiJoin-Tpcds.mdp]: I don't really know, it's allegedly using the legacy cost model and disabling a transform, so I don't know the intent. Do we think it's even worth keeping?

**Comment by chrishajas** [comment]
> Thanks for finding this and opening a PR, I'm OK with removing these - especially the unused tests!
> 
> Any idea why the files were added in the first place?
> 
> * data/dxl/expressiontests
> * data/dxl/plstmt
> * data/dxl/query
> * data/dxl/tpch-partitioned/
>   etc.
>   I know some of these tests were disabled because it was thought to not provide meaningful failures as MDPs (esp regarding tpch tests).
> 
> Shouldn't we keep the files in data/dxl/minidump like:
> 
> * data/dxl/minidump/Join-Varchar-Equality.mdp
> * data/dxl/minidump/ThreeStageAgg-DistinctOnComputedCol.mdp
>   ?

@hardikar 

Some of the test drivers that ran these tests were lost in the move to cmake. Many of them were added back in 2011 in large groups, most likely just enhancing test coverage. My thinking is that since they haven't been run/touched in 5+ years, there isn't interest in maintaining them.

I'm adding back the files in the minidump directory. They don't have the associated query in the comments, but they do run so we may as well add them back.

**Pull request merged**